frontend/src/components/modal-compare/modal.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, Dispatch } from 'react';
+
+import fillCompare from '@assets/images/icon_balance.svg';
+
+import styles from './styles.module.scss';
+
+interface ModalProps {
+  carName: string;
+  carDescription: string;
+  isHidden: boolean;
+  setIsHidden: Dispatch<boolean>;
+}
+
+const Modal: FC<ModalProps> = ({
+  carName,
+  carDescription,
+  isHidden,
+  setIsHidden,
+}) => {
+  const closeModal = (): void => {
+    setIsHidden(!isHidden);
+  };
+
+  return (
+    <div className={styles.container} hidden={isHidden}>
+      <div className={styles.content}>
+        <button className={styles.close} onClick={(): void => closeModal()}>
+          ×
+        </button>
+        <div className={styles.icon}>
+          <img
+            className={styles.button}
+            src={fillCompare}
+            alt="compare button"
+          />
```

Comment:
**`img` cannot be the button**. And `div` cannot be a button. Only the `button` tag can be the button :slightly_smiling_face:. And the `button` can wrap the `img`